packages/core/src/readers/LlamaParseReader.ts

```diff
@@ -0,0 +1,110 @@
+import { Document } from "../Node";
+import { defaultFS } from "../env";
+import { GenericFileSystem } from "../storage/FileSystem";
+import { FileReader } from "./type";
+
+type ResultType = "text" | "markdown";
+
+/**
+ * Represents a reader for parsing files using the LlamaParse API.
+ * See https://github.com/run-llama/llama_parse
+ */
+export class LlamaParseReader implements FileReader {
+  // The API key for the LlamaParse API.
+  apiKey: string;
+  // The base URL of the Llama Parsing API.
+  baseUrl: string = "https://api.cloud.llamaindex.ai/api/parsing";
+  // The maximum timeout in seconds to wait for the parsing to finish.
+  maxTimeout = 2000;
+  // The interval in seconds to check if the parsing is done.
+  checkInterval = 1;
+  // Whether to print the progress of the parsing.
+  verbose = true;
+  resultType: ResultType = "text";
+
+  constructor(params: Partial<LlamaParseReader> = {}) {
+    Object.assign(this, params);
+    params.apiKey = params.apiKey ?? process.env.LLAMA_CLOUD_API_KEY;
+    if (!params.apiKey) {
+      throw new Error(
+        "API Key is required for LlamaParseReader. Please pass the apiKey parameter or set the LLAMA_CLOUD_API_KEY environment variable.",
+      );
+    }
+    this.apiKey = params.apiKey;
+  }
+
+  async loadData(
+    file: string,
+    fs: GenericFileSystem = defaultFS,
+  ): Promise<Document[]> {
+    if (!file.endsWith(".pdf")) {
+      throw new Error("Currently, only PDF files are supported.");
+    }
+
+    const metadata = { file_path: file };
+
+    // Load data, set the mime type
+    const data = await fs.readRawFile(file);
+    const body = new FormData();
+    body.set("file", new Blob([data], { type: "application/pdf" }), file);
```

Comment:
@hexapode this code didn't work without explicitly setting the mime-type - maybe the API could guess the mime type?